checks.d/kubernetes_topology.py

```diff
@@ -64,46 +67,73 @@ def _extract_nodes(self, kubeutil, instance_key):
+
+            deployment_template = deployment['spec']['template']
+            if deployment_template and deployment_template['metadata']['labels']:
+                data['template_labels'] = self._make_labels(kubeutil, deployment_template['metadata'])
+                replicasets = kubeutil.retrieve_replicaset_filtered_list(deployment['metadata']['namespace'], deployment_template['metadata']['labels'])
+                if replicasets['items']:
+                    for replicaset in replicasets['items']:
+                        self.relation(instance_key, externalId, replicaset['metadata']['name'], {'name': 'CREATED'}, dict())
+
+            self.component(instance_key, externalId, {'name': 'KUBERNETES_DEPLOYMENT'}, data)
+
     def _extract_pods(self, kubeutil, instance_key):
-        replicasets = defaultdict(list)
+        replicasets_to_pods = defaultdict(list)
+        replicaset_to_data = dict()
         for pod in kubeutil.retrieve_pods_list()['items']:
             data = dict()
             pod_name = pod['metadata']['name']
             data['uid'] = pod['metadata']['uid']
-            data['labels'] = self._flatten_dict(kubeutil.extract_metadata_labels(pod['metadata']))
+            data['namespace'] = pod['metadata']['namespace']
+            data['labels'] = self._make_labels(kubeutil, pod['metadata'])
 
             self.component(instance_key, pod_name, {'name': 'KUBERNETES_POD'}, data)
 
             relation_data = dict()
-            self.relation(instance_key, pod_name, pod['spec']['nodeName'], {'name': 'HOSTED_ON'}, relation_data)
+            self.relation(instance_key, pod_name, pod['spec']['nodeName'], {'name': 'PLACED_ON'}, relation_data)
 
             if 'containerStatuses' in pod['status'].keys():
-                self._extract_containers(instance_key, pod_name, pod['status']['podIP'], pod['status']['hostIP'], pod['spec']['nodeName'], pod['status']['containerStatuses'])
+                self._extract_containers(instance_key, pod_name, pod['status']['podIP'], pod['status']['hostIP'], pod['spec']['nodeName'], pod['metadata']['namespace'], pod['status']['containerStatuses'])
 
             if 'ownerReferences' in pod['metadata'].keys():
                 for reference in pod['metadata']['ownerReferences']:
                     if reference['kind'] == 'ReplicaSet':
                         data = dict()
                         data['name'] = pod_name
-                        replicasets[reference['name']].append(data)
-
-        for replicaset_name in replicasets:
-            self.component(instance_key, replicaset_name, {'name': 'KUBERNETES_REPLICASET'}, dict())
-            for pod in replicasets[replicaset_name]:
-                self.relation(instance_key, pod['name'], replicaset_name, {'name': 'CONTROLLED_BY'}, dict())
-
-    def _extract_containers(self, instance_key, pod_name, pod_ip, host_ip, host_name, statuses):
+                        replicasets_to_pods[reference['name']].append(data)
```

Comment:
Explanation and a link with what replicasets are would be nice here, in a comment